dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2119,65 @@ def _update_chi2(self):
 
         return chi2_mat
 
+    def _calc_nan_replication_dependent_properties(self, clean_samples):
+        """
+        Calculate metrics needed for replicating NaN values in capitalone/synthetic-data.
+
+        Metrics include class priors and class specific means of each column.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        """
+        if not self.options.synthetic_data.replicate_nan.is_enabled:
+            return
+
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        for col_id in range(len(self._profile)):
+            null_count = getattr(self._profile[col_id], "null_count")
+            if null_count == 0:
+                # No missing values to replicate
+                continue
+
+            # Partition data based on whether target column value is NaN or not
+            col_is_nan = data.iloc[:, col_id].isnull()
+            partition_nan = data[col_is_nan]
+            partition_not_nan = data[~col_is_nan]
+
+            # Drop the target column from partitions
+            partition_nan = partition_nan.drop(columns=col_id)
+            partition_not_nan = partition_not_nan.drop(columns=col_id)
+
+            # Calculate mean vectors of each partition
+            # i.e mean vectors of rows where target col is or is not NaN
+            mean_nan, mean_not_nan = (
+                partition_nan.mean().to_numpy(),
+                partition_not_nan.mean().to_numpy(),
+            )
+
+            # Calculate class priors
+            # i.e probability that a value in target col is or is not NaN
+            sample_size = getattr(self._profile[col_id], "sample_size")
+            true_count = sample_size - null_count
+
+            prior_nan = null_count / sample_size
+            prior_not_nan = true_count / sample_size
+
+            col_name = getattr(self._profile[col_id], "name")
+            self._nan_replication_dependent_properties.append(
+                {
+                    "column_id": col_id,
+                    "column_name": col_name,
+                    # Array index serves as class label
+                    # 0 indicates not NaN, 1 indicates NaN
+                    "class_priors": [prior_not_nan, prior_nan],
+                    "class_mean": [mean_not_nan, mean_nan],
+                }
+            )
```

Comment:
how do we handle this if we update the profile? wouldn't this be  incorrect/

Reply:
Yep working on making it handle updates.